gadopt/approximations.py

```diff
@@ -413,12 +421,66 @@ def buoyancy(self, p, T):
 
 
 class SmallDisplacementViscoelasticApproximation:
-    """Expressions for the small displacement viscoelastic approximation.
+    """Base class for viscoelasticity assuming small displacements.
+
+    By assuming a small displacement with respect to mantle depth
+    we can linearise the problem, assuming a perturbation away from a reference state.
+
+    For background and derivation of the formulation please see the equations and
+    references provided in Scott et al 2025.
+
+    Automated forward and adjoint modelling of viscoelastic deformation of the solid
+    Earth.  Scott, W.; Hoggard, M.; Duvernay, T.; Ghelichkhan, S.; Gibson, A.;
+    Roberts, D.; Kramer, S. C.; and Davies, D. R. EGUsphere, 2025: 1–43. 2025.
+
+    Arguments:
+      density:       density of the reference state - assumed to be hydrostatic
+      shear_modulus: shear modulus
+      viscosity:     viscosity
+      g:             gravitational acceleration
+      B_mu:          Nondimensional number describing ratio of buoyancy to elastic
+                     shear strength used for nondimensionalisation.
+                     $$ B_{\\mu} = \frac{\bar{\rho} \bar{g} L}{\bar{\\mu}}$,
+                     where $\bar{\rho}$ is a characteristic density scale (kg / m^3),
+                     $\bar{g}$ is a characteristic gravity scale (m / s^2),
+                     $L$ is a characteristic length scale, often Mantle depth (m),
+                     $\\mu$ is a characteristic shear modulus (Pa).
+    """
+
+    def __init__(
+        self,
+        density: Function | Number,
+        shear_modulus: Function | Number | list,
+        viscosity: Function | Number | list,
+        *,
+        g: Function | Number = 1,
+        B_mu: Function | Number = 1,
+    ):
+        self.density = ensure_constant(density)
+        self.shear_modulus = ensure_constant(shear_modulus)
+        self.viscosity = ensure_constant(viscosity)
+        self.g = ensure_constant(g)
+        self.B_mu = ensure_constant(B_mu)
+
+    def buoyancy(self, displacement):
+        # Buoyancy term due to the advection of the background density field
+        # written on RHS of equations
+        return -self.B_mu * self.g * -inner(displacement, grad(self.density))
```

Comment:
Any reason to keep two `-` signs?

Reply:
Not really I guess in my head I was still thinking of the term as `-B_mu * g * rho_1`  where `rho_1 = -grad u \cdot grad rho_0`

Reply:
removed minus signs